pdr_backend/accuracy/app.py

```diff
@@ -350,12 +354,40 @@ def serve_statistics_from_file():
     If the file cannot be read or another error occurs, it returns a 500 Internal Server Error.
     """
 
+    four_weeks_ago = datetime.utcnow() - timedelta(weeks=4)
+    start_ts = UnixTimeS(int(four_weeks_ago.timestamp()))
+    end_ts = UnixTimeS(int(datetime.utcnow().timestamp()))
     try:
-        with open(JSON_FILE_PATH, "r") as f:
-            data = json.load(f)
-            response = jsonify(data)
-            response.headers.add("Access-Control-Allow-Origin", "*")  # Allow any origin
-            return response
+        db_conn = DuckDBDataStore("./lake_data", read_only=True)
+        slots_table_name = Slot.get_lake_table_name()
+        slots_table = db_conn.query_data(
+            f"""
+            SELECT * FROM {slots_table_name} WHERE SLOT > {start_ts} AND SLOT < {end_ts}
+            """
+        )
+        db_conn.duckdb_conn.close()
+
+        slots_table = slots_table.group_by("ID").first()
+        slots_table = slots_table.sort("slot")
+
+        all_slots: List[Slot] = []
+
+        # Iterate over rows and create objects
+        for row in slots_table.rows(named=True):
+            slot = Slot(
+                row["ID"],
+                row["timestamp"],
+                row["slot"],
+                row["truevalue"],
+                row["roundSumStakesUp"],
+                row["roundSumStakes"],
+            )
+            all_slots.append(slot)
+
+        data = transform_slots_to_statistics(all_slots)
+        response = jsonify(data)
+        response.headers.add("Access-Control-Allow-Origin", "*")  # Allow any origin
+        return response
     except Exception as e:
         response = jsonify({"error": "Internal Server Error", "message": str(e)})
```

Comment:
not all exceptions are created equal. Maybe not now, but in the future, we want to be more specific with our exceptions. E.g. if no results are found, then it's a 404. Same with other codes.

Reply:
Same as before. I agree.

I have documented this in [#1424](https://github.com/oceanprotocol/pdr-backend/issues/1424)